src/utils/actions.js

```diff
@@ -0,0 +1,254 @@
+              // get value by param
+              const date = getBySearchParam(resource, searchParam); // assume its a datetime?
+              return compareDates(date, filterPeriod);
+            });
+            filteredResources.push.apply(filteredResources, dateFiltered);
+          }
+        } else {
+          // if there's no date filter, we can just push all the resources
+          // in to be filtered by code
+          filteredResources.push.apply(filteredResources, resources);
+        }
+
+        const codeFilter = input.codeFilter;
+        if (codeFilter) {
+          // code filter isn't under the same constraint as datefilter
+          // we can assume that it's just a path
+          const path = codefilter.path;
+          const searchParam = codeFilter.searchParam;
+          if (path) {
+            // no need to refilter resources
+            const filteredResources = filteredResources.filter(resource => {
+              const code = getByPath(resource, path);
+              return compareCodes(code, codeFilter.code); // codeFilter.code is of type Coding, could also be a valueSet?
+            });
+          } else if (searchParam) {
+            const filteredResources = filteredResources.filter(resource => {
+              const code = getBySearchParam(resource, searchParam);
+              return compareCodes(code, codeFilter.code);
+            });
+          }
+        }
+        return context.records.push(filteredResources);
+      });
+    }
+  },
+  'execute-research-query': context => {
+    // noop
+  }
+};
+
+function validateProfile(report, structureDefinition) {
+  // TODO: Perhaps this is something one of the servers/clients would handle?
+  return report;
+}
+
+function getByPath(resource, path) {
+  // TODO: function to search resources by path
+  // and return correct attribute
+  return resource;
+}
+
+function compareCodes(coding1, coding2) {
+  // assume they're both coding
+  return coding1.code === coding2.code && coding1.system === coding2.system;
+}
+
+function getBySearchParam(resource, searchParam) {
+  // TODO: function to search resources by param
+  // and return correct attribute
+  return resource.date;
+}
+
+function compareDates(period1, period2) {
+  // TODO: returns true if the periods overlap
+}
+function createBundle(records) {
+  const bundle = {
+    resourceType: 'Bundle',
+    type: 'collection',
+    entry: []
+  };
+
+  records.forEach(record => {
+    bundle.push({
+      resource: record
+    });
+  });
+
+  return bundle;
+}
+
+function getPatient(context) {
+  const patientId = context.patient;
+  const patient = context.client.source.read(patientId);
+  context.patient = patient;
+  return context;
+}
+
+function evaluateExpression(expression, resource) {
+  if (expression.language === 'text/fhirpath') {
+    const fhirpath = new FhirPath(resource);
+    const path = fhirpath.evaluate(expression.expression);
+    return path;
+  }
+}
+
+function makeHeader(profile) {
```

Comment:
This is missing some important elements of the [Public Health MessageHeader](https://build.fhir.org/ig/HL7/fhir-medmorph/StructureDefinition-us-ph-messageheader.html), such as source and destination.